File: core/client/game/src/commonGame/net/base/GameRequest.cs
using ShineEngine;

/// <summary>
/// 逻辑服Request
/// </summary>
public class GameRequest:BaseRequest
{
	public GameRequest()
	{
		//客户端request
		setNeedFullWrite(ShineSetting.clientMessageUseFull);
	}

	/// <summary>
	/// 发送
	/// </summary>
	public void send()
	{
		send(true);
	}
	
	/// <summary>
	/// 发送
	/// </summary>
	public void send(bool needLog)
	{
		if(CommonSetting.isSingleGame)
			return;

		GameC.server.getSocket().send(this,needLog);
	}
}

File: core/clientLaya/game/src/commonGame/net/base/GameRequest.ts
namespace Shine
{
	export class GameRequest extends BaseRequest
	{
	    public constructor()
		{
			super();
			//客户端request
			this.setOpenCheck(true);
		}
	
		/** 发送 */
		public send():void
		{
			GameC.server.getSocket().send(this);
		}
	}
}